core/src/tile/tileWorker.cpp

```diff
@@ -64,43 +63,51 @@ void TileWorker::run() {
             // Pop highest priority tile from queue
             auto it = std::min_element(m_queue.begin(), m_queue.end(),
                 [](const auto& a, const auto& b) {
-                    if (a->tile->isVisible() != b->tile->isVisible()) {
-                        return a->tile->isVisible();
+                    if (a->isProxy() != b->isProxy()) {
+                        return !a->isProxy();
                     }
-                    if (a->tile->sourceID() == b->tile->sourceID() &&
-                        a->tile->sourceGeneration() != b->tile->sourceGeneration()) {
-                        return a->tile->sourceGeneration() < b->tile->sourceGeneration();
+                    if (a->source().id() == b->source().id() &&
+                        a->sourceGeneration() != b->sourceGeneration()) {
+                        return a->sourceGeneration() < b->sourceGeneration();
```

Comment:
This order ensures that the earlier requests are processed first. Otherwise when there are continuous updates to the source the lower prioritized tiles would never be updated.